src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellSearchViewAdapter.cs

```diff
@@ -71,7 +71,13 @@ DataTemplate DefaultTemplate
 						if (RuntimeFeature.IsShellSearchResultsRendererDisplayMemberNameSupported)
 						{
 #pragma warning disable CS0618
+#if NET8_0
+#pragma warning disable IL2026 // FeatureGuardAttribute is not supported on .NET 8
+#endif
 							label.SetBinding(Label.TextProperty, _searchHandler.DisplayMemberName ?? ".");
```

Comment:
I think the feature flag is the linker magic to replace the value in the property and cause this path to be linked out (or not). But in .NET 8, this magical linker does not exist, so the flag will always be there and the code path will always exist and potentially run if the switch is set.

For example, the case where the `IsIVisualAssemblyScanningEnabled` flag is set, in net8 this always existed and always ran since that was the code. In .net 9 we wrap it in a feature flag that is controlled by the `Microsoft.Maui.RuntimeFeature.IsIVisualAssemblyScanningEnabled` app context switch. Since this magic to remove the branch if it is not set does not exist, it will always be there. If it _is_ set, it will still be there, but the branch will not run.

Not as amazing as net9, but net8 can technically use the same switches and get the same behavior as net9 - just without the linker magic trimming the code. However, in all cases, this should not affect anything for net8 since that code always was there in net8 and was never trimmed.  Now we got the "bonus" of net9 behaviour without the smaller assemebly trimming results.